### ISSUE TITLE
[Translatable] Using a personnal query result in translation bug (Array to string conversion)

### ISSUE THREAD

**Issue by homersimpsons** [KnpLabs/DoctrineBehaviors#344]
Hello,

I'm using Translatable to get translation of database Item.
I created a personnal query to fetch the latest events:
`$queryBuilder->where('e.startTime > :date')
					->setParameter('date', new \DateTime())
				 ->orderBy('e.startTime', 'DESC')
				 ->setMaxResults($limit);
$query = $queryBuilder->getQuery();
$results = $query->getArrayResult();
return $results;`
And when I'm using a twig filter to get the right translation (`{{ event.translate(app.request.locale).name }}`) I've got the error message:

> An exception has been thrown during the rendering of a template ("Notice: Array to string conversion")

Note that change `getArrayResult()` to `getResult()` doesn't do anything.
Note that I tried to do:
`foreach($nextEvents as $event){
	$event = $event->getTranslation($request->getLocale());
}`
in Controller, but I've got an error too:

> Error: Call to a member function getTranslation() on array
(with both `getArrayResult()` and `getResult()`)

I hope you understand the problem.
Don't hesitate to reply for more informations.

**Comment by docteurklein** [comment]
Could you paste here the entire stack trace ? At the moment I don't see what could cause the problem.

**Comment by homersimpsons** [comment]
Up: I think that I had to use [attribute](http://twig.sensiolabs.org/doc/functions/attribute.html)
like that: `{{ attribute(event, translate(app.request.locale)~'.name' }}` (or something like that) because I build my own translation system using database column, and I face this problem too.
It's hard for twig to interpret complex syntax like `{{ event.translate(app.request.locale).name }}`
Where:
- `event` is an object
- `translate()` is a part of that object linked to other ressources
- `app.request.locale` return a string extracted from another object
- `name` return a string from the object `event.translate()` (getName)

Will try to provide a stack Trace, but I created another branch to try something else. Then I need to reimport KnpLabs/DoctrineBehaviors(Translatable) via composer and recreate the database.

But As I said, using [attribute](http://twig.sensiolabs.org/doc/functions/attribute.html) should fix my problem that is not KnpLabs/DoctrineBehaviors(Translatable) related.
